### DISCUSSION TITLE
Missing some powerline font symbols

### DISCUSSION THREAD

**Discussion by aleksanderd** [starship/starship#1168] [General]
Installed fish & starship at fresh MxLinux 19.1 and noticed missing some glyphs(at least for php and python) while others ones ok.

![image](https://user-images.githubusercontent.com/5766592/80961599-f427e700-8e13-11ea-9c49-1efb5eff0717.png)

tried already:
1. Install/reinstall powerline fonts
2. Latest FiraCode 3.1
3. Patched FiraMono from powerline
4. fc-cache -rvf after any changes

It is strange the most icons which is ok, are ok for ANY font, even without powerline fonts installed... But php/python icons missing even if using patched font.

What else to try?

Thank you!

**Comment by davidkna** [comment]
Is it possible the icons got corrupted when copy/pasting the powerline config? Have you tried downloading it directly or re-copying it now that you have the fonts installed?

**Reply by aleksanderd** [reply]
I compared installed(via apt) powerline files with ones downloaded directly from the repo - no diffs...
FiraCode and FiraMono are downloaded from the repos directly already...

what concrete you mean to re-download?

**Reply by davidkna** [reply]
Sorry I misread and thought you were using the nerd font config preset.
The default PHP and Python symbols are emoji. Do you have an emoji font installed? For example the noto emoji font should work.

**Reply by aleksanderd** [reply]
Huge thanks! 

`apt install fonts-noto-color-emoji` solve the problem.

What about to add this to docs/readme? ;)

**Reply by chipbuster** [reply]
Based on the number of issues we've been getting due to lack of emoji support on terminal platforms, I think we do need to document this somehow. However, since the install is platform-specific (e.g. on Arch it's `pacman -S noto-fonts-emoji`), we might not want to post platform-specific install instructions.

Does anyone know if there's a README/blog post out there that describes how to install emoji fonts on various linux distros?

**Reply by aleksanderd** [reply]
I mean add just a word about emoji, w/o any platform specific things... 

Currently there is one line at  Prerequisites:
`A Powerline font installed and enabled in your terminal (for example, try Fira Code ).`

And I believed this is enough info so even not tried to play with something other than powerline...
